tracer/test/Datadog.Trace.Tests/Propagators/MultiSpanContextPropagatorTests.cs

```diff
@@ -0,0 +1,272 @@
+// <copyright file="MultiSpanContextPropagatorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Headers;
+using Datadog.Trace.Propagators;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Propagators
+{
+    public class MultiSpanContextPropagatorTests
+    {
+        private static readonly SpanContextPropagator Propagator;
+
+        static MultiSpanContextPropagatorTests()
+        {
+            var names = new[]
```

Comment:
Maybe you could enumerate the enum values here, that way if we add a new propagation type it will automatically be called in these tests. What do you think?

Reply:
I have to think about it, because is not only the enumeration, but adding it in each test case as well...

Reply:
I was thinking that if you add the enumerated enums and we add a new propagator, then the `headers.VerifyNoOtherCalls()` call in the `Inject_All_IHeadersCollection` should fail, which would be an easy reminder that we need to add test cases here

Reply:
Oh! That’s a good one, will do it!

Reply:
done in [f0367d7](https://github.com/DataDog/dd-trace-dotnet/pull/2536/commits/f0367d7e292c04564d0fbc50584334191ba36d73)